fbw-common/src/wasm/systems/systems/src/air_conditioning/cabin_pressure_controller.rs

```diff
@@ -575,29 +644,39 @@ impl<C: PressurizationConstants> ControllerSignal<PressureValveSignal>
 
 impl<C: PressurizationConstants> SimulationElement for CabinPressureController<C> {
     fn write(&self, writer: &mut SimulatorWriter) {
-        // Add check for active cpc only
-        writer.write(&self.cabin_altitude_id, self.cabin_altitude());
-        writer.write(
-            &self.outflow_valve_open_percentage_id,
-            self.outflow_valve_open_amount,
-        );
-        writer.write(
-            &self.safety_valve_open_percentage_id,
-            self.safety_valve_open_amount,
-        );
-        writer.write(
-            &self.cabin_vs_id,
-            self.cabin_vertical_speed().get::<foot_per_minute>(),
-        );
-        writer.write(&self.cabin_delta_pressure_id, self.cabin_delta_p());
-
-        // FWC warning signals
-        writer.write(&self.fwc_excess_cabin_altitude_id, self.is_excessive_alt());
-        writer.write(
-            &self.fwc_excess_residual_pressure_id,
-            self.is_excessive_residual_pressure(),
-        );
-        writer.write(&self.fwc_low_diff_pressure_id, self.is_low_diff_pressure());
+        let ssm = if self.is_active {
+            SignStatus::NormalOperation
+        } else if self.failure.is_active() {
+            SignStatus::FailureWarning
+        } else {
+            SignStatus::NoComputedData
+        };
+
+        if self.is_active {
+            // Safety valve open percentage is not sent by the CPC in real life
+            writer.write(
+                &self.safety_valve_open_percentage_id,
+                self.safety_valve_open_amount,
+            );
+
+            writer.write_arinc429(&self.cabin_delta_pressure_id, self.cabin_delta_p_out(), ssm);
+            writer.write_arinc429(&self.cabin_altitude_id, self.cabin_altitude_out(), ssm);
+            writer.write_arinc429(
+                &self.outflow_valve_open_percentage_id,
+                self.outflow_valve_open_amount_out(),
+                ssm,
+            );
+            writer.write_arinc429(
+                &self.cabin_vs_id,
+                self.cabin_vertical_speed_out().get::<foot_per_minute>(),
+                ssm,
+            );
+            writer.write_arinc429(
+                &self.landing_elevation_id,
+                self.landing_elevation_out(),
+                ssm,
+            );
```

Comment:
You probably want to write these values even if `self.is_active` is false else the ssm will always be normal operation. Or do I miss something?

Reply:
Since we have one ARINC variable for both of the CPCs, I wanted to avoid CPC 2 overwriting the values with `NoComputedData` when CPC1 was active. But you are right that this always returned `NormalOperation` because the conditional was incorrect. I've changed it now so if both CPCs fail the SSM is `FailureWarning`.

In reality both CPCs should be sending separate ARINC words, and these "cleaned" within the SDAC depending on which system is active. As we don't have the SDAC modelled yet this can be left for future work imo as we probably don't want to add all of the logic into the display js code.

Reply:
I understand what you're doing here. Just generally it's a bad idea to write a simvar from 2 different systems.

Reply:
> In reality both CPCs should be sending separate ARINC words, and these "cleaned" within the SDAC depending on which system is active

This is actually not correct, both SDACs retransmit the A429 words from both CPCs, under different SDIs

Reply:
So how do we proceed?

Reply:
The CPCs should write their own separate vars. It's not their job to implement logic for other systems, and IMO it's better to create a stub for the other system to do it if needed.

Reply:
I can make the CPCs write separate vars that's no problem. The issue is all the logic to select which one to use will end up in the javascript display end, but I'll implement this 👍

Reply:
We have much better tools to deal with data from the MSFS SDK on the typescript side, so this is a positive IMO.

Reply:
Fixed now, each CPC sends its own arinc vars and the instruments select based on which CPC is active. 👍